packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -344,8 +344,25 @@ export function buildThunks<
         }
       )
     } catch (error) {
-      if (error instanceof HandledError) {
-        return rejectWithValue(error.value, { baseQueryMeta: error.meta })
+      let catchedError = error
+      if (catchedError instanceof HandledError) {
+        let transformErrorResponse: (
+            baseQueryReturnValue: any,
+            meta: any,
+            arg: any
+        ) => any = defaultTransformResponse
+
+        if (endpointDefinition.query && endpointDefinition.transformErrorResponse) {
+          transformErrorResponse = endpointDefinition.transformErrorResponse
+        }
+        try {
+          return rejectWithValue(
+              await transformErrorResponse(catchedError.value, catchedError.meta, arg.originalArgs),
+              { baseQueryMeta: catchedError.meta }
+          )
+        } catch (e) {
+          catchedError = e
```

Comment:
is it an ok thing to do? didn't want to repeat the whole console.error/throw block here